### PR TITLE
AWS Lambda documentation

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> Documentation for the AWS Lambda destination

## Type of documentation

- [x] New documentation
- [ ] Documentation update
- [ ] Hotfix

### CODE REVIEW

**Comment by ameypv-rudder** [comment]
This PR is on hold as per discussion with @sandeepdsvs. cc @ishachopra0401 @utsabc

File: docs/destinations/streaming-destinations/aws-lambda.mdx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+| Function name | <ul><li>`lambda-function` (Name only)</li><li>`lambda-function:v1` (Name with alias)</li></ul>|
+| Function ARN | `arn:aws:lambda:us-west-2:123456789012:function:my-function` |
+| Partial ARN | `123456789012:function:lambda-function` |
+
+<div class="successBlock">
+You can also append a version number or alias to any of the above formats.
+</div>
+
+- **Invocation Type**: This setting lets you specify how RudderStack invokes the lambda function. You can choose between any of the following two options in the dropdown:
+  - **Event**: RudderStack invokes the lambda function asynchronously and sends the failed events to the function's dead-letter queue (if configured). For more information, refer to the <a href="https://docs.aws.amazon.com/lambda/latest/dg/invocation-async.html">AWS Lambda documentation</a>.
```
Comment (utsabc) [inline docs/destinations/streaming-destinations/aws-lambda.mdx:60]: Lets change this to `Async Invocation`
and `Sync Invocation`
@sandeepdsvs

**Review by utsabc — CHANGES_REQUESTED** [review]

File: docs/destinations/streaming-destinations/aws-lambda.mdx (hunk 1 above)
Comment (sandeepdsvs) [reply docs/destinations/streaming-destinations/aws-lambda.mdx]: No longer using this setting.

File: docs/destinations/streaming-destinations/aws-lambda.mdx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+For more information on the AWS regions, refer to the <a href="https://aws.amazon.com/about-aws/global-infrastructure/regions_az/">AWS Regions and Availability Zones</a> guide.
+</div>
+
+- **Role Based Authentication**: Enable this setting to use role-based AWS authentication for this destination. If this setting is enabled, you can configure the following fields:
+  <img src="../../assets/event-stream-destinations/aws-lambda-connection-settings-2.png" alt="AWS Lambda connection settings"/>
+
+  - **IAM Role ARN**: Enter your role ARN. For more information on the IAM ARNs, refer to the <a href="https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_identifiers.html#identifiers-arns">AWS documentation</a>.
+  - **External ID**: Enter the external ID associated with the role. For more information on using the external ID, refer to the <a href="https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html">AWS documentation</a>.
+- **Access Key ID**: Enter the AWS access key ID associated with your IAM user account.
+- **Secret Access Key**: Enter your secret access key.
```
Comment (sandeepdsvs) [inline docs/destinations/streaming-destinations/aws-lambda.mdx:46]: @ameypv-rudder  Don't we need to mention `Access Key ID` and `Secret Access Key` will be disabled if `Role Based Authentication` ?

Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/aws-lambda.mdx]: thanks - I've incorporated this feedback.

File: docs/destinations/streaming-destinations/aws-lambda.mdx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+  - **IAM Role ARN**: Enter your role ARN. For more information on the IAM ARNs, refer to the <a href="https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_identifiers.html#identifiers-arns">AWS documentation</a>.
+  - **External ID**: Enter the external ID associated with the role. For more information on using the external ID, refer to the <a href="https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html">AWS documentation</a>.
+
+If **Role Based Authentication** is disabled, you need to configure the following settings:
+  - **Access Key ID**: Enter the AWS access key ID associated with your IAM user account.
+  - **Secret Access Key**: Enter your secret access key.
+
+<div class="infoBlock">
+For more information on creating an IAM user in AWS, refer to the <a href="https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html">AWS documentation</a>.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/aws-lambda.mdx:51]:
```suggestion
For more information on creating an IAM user in AWS, refer to the <a href="https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_common-scenarios_third-party.html">AWS documentation</a>.
```

Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/aws-lambda.mdx:51]: @sandeepdsvs please verify if this update is accurate.

Comment (sandeepdsvs) [reply docs/destinations/streaming-destinations/aws-lambda.mdx:51]:
```suggestion
For more information on creating an IAM role in AWS, refer to the <a href="https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_common-scenarios_third-party.html">AWS documentation</a>.
```

File: docs/destinations/streaming-destinations/aws-lambda.mdx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+<div class="infoBlock">
+It is highly recommended to use the <Link to="/resources/glossary/#workspace-token">workspace token</Link> as the external ID while creating the IAM role.
+</div>
+
+If **Role Based Authentication** is disabled, you need to configure the following settings:
+  - **Access Key ID**: Enter the AWS access key ID associated with your IAM user account.
+  - **Secret Access Key**: Enter your secret access key.
+
+<div class="infoBlock">
+For more information on creating an IAM user in AWS, refer to the <a href="https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_common-scenarios_third-party.html">AWS documentation</a>.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/aws-lambda.mdx:55]:
```suggestion
For more information on creating an IAM role in AWS, refer to the <a href="https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_common-scenarios_third-party.html">AWS documentation</a>.
```

Comment (sandeepdsvs) [reply docs/destinations/streaming-destinations/aws-lambda.mdx]: 👍🏼

**Review by sandeepdsvs — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Review by ishachopra0401 — APPROVED** [review]

**Pull request merged**